client/src/main/scala/org/http4s/client/middleware/FollowRedirect.scala

```diff
@@ -23,23 +21,36 @@ object FollowRedirect {
                 fragment = uri.fragment orElse req.uri.fragment
               )
 
-              prepareLoop(
-                req.copy(method = method, uri = nextUri, body = EmptyBody),
-                redirects + 1
-              )
+              // We're following a redirect, so we need to dispose of the last response
+              dispose.flatMap { _ =>
+                prepareLoop(
+                  req.copy(method = method, uri = nextUri, body = EmptyBody),
+                  redirects + 1
+                )
+              }
 
             case _ => Task.now(dr)
           }
         }
 
         resp.status.code match {
-          // We cannot be sure what will happen to the request body so we don't attempt to deal with it
-          case 301 | 302 | 307 | 308 if req.body.isHalt =>
-            dispose.flatMap(_ => doRedirect(req.method))
+          case 301 | 302 =>
+            doRedirect(req.method match {
+              case Method.POST => Method.GET
```

Comment:
I see this is allowed by the spec, but is it a good idea? It is said to be legal for "historical reasons." 
https://tools.ietf.org/html/rfc7231#section-6.4.2